src/generic/MatRing.jl

```diff
@@ -16,27 +16,46 @@ elem_type(::Type{MatRing{T}}) where {T <: NCRingElement} = MatRingElem{T}
 
 base_ring(a::MatRing{T}) where {T <: NCRingElement} = a.base_ring::parent_type(T)
 
+base_ring(a::MatRingElem{T}) where {T <: NCRingElement} = base_ring(matrix(a))
+
 @doc raw"""
     parent(a::MatRingElem{T}) where T <: NCRingElement
 
 Return the parent object of the given matrix.
 """
-parent(a::MatRingElem{T}) where T <: NCRingElement = MatRing{T}(base_ring(a), size(a.entries)[1])
+parent(a::MatRingElem{T}) where T <: NCRingElement = MatRing{T}(base_ring(a), nrows(matrix(a)))
 
 is_exact_type(::Type{MatRingElem{T}}) where T <: NCRingElement = is_exact_type(T)
 
 is_domain_type(::Type{MatRingElem{T}}) where T <: NCRingElement = false
 
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+number_of_rows(a::MatRingElem) = nrows(matrix(a))
+
+number_of_columns(a::MatRingElem) = ncols(matrix(a))
+
+Base.@propagate_inbounds getindex(a::MatRingElem, r::Int, c::Int) = matrix(a)[r, c]
+
+Base.@propagate_inbounds function setindex!(a::MatRingElem, d::NCRingElement,
+                                            r::Int, c::Int)
+    matrix(a)[r, c] = base_ring(a)(d)
```

Comment:
```suggestion
    matrix(a)[r, c] = d
```
I would have expected that `setindex!` for the underlying matrix element takes care of this.